common/loglevel.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+// #ifndef MENDER_LOGLEVEL_HPP
+// #define MENDER_LOGLEVEL_HPP
+
+#include <cassert>
+
+namespace mender::log {
+
+using namespace std;
+
+enum class LogLevel {
+	Fatal = 5,
+	Error = 4,
+	Warning = 3,
+	Info = 2,
+	Debug = 1,
+	Trace = 0,
+};
+
+inline std::string to_string_level(LogLevel lvl) {
+	switch (lvl) {
+	case LogLevel::Fatal:
+		return "fatal";
+	case LogLevel::Error:
+		return "error";
+	case LogLevel::Warning:
+		return "warning";
+	case LogLevel::Info:
+		return "info";
+	case LogLevel::Debug:
+		return "debug";
+	case LogLevel::Trace:
+		return "trace";
+	}
```

Comment:
There must be a better way to do this in C++

Reply:
Yeh, this and the formatter has avoided my attempts at abstraction so far, so I just kept it simple to get it done in the first instance :see_no_evil:  :seedling: